### ISSUE TITLE
API Proposal: Path API that exposes the Portable Filename Character Set

### ISSUE THREAD

**Issue by JeremyKuhne** [dotnet/corefx#27048]
``` c#
namespace System.IO
{
    public static class Path
    {
        public static bool IsPortableFileName(ReadonlySpan<char> name);
    }
}
```

Returns true if the name only has characters from the defined set. The set is small `[A-Z][a-z][0-9]._-`.

POSIX.1-2008
3.282 Portable Filename Character Set
http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_282

**Comment by danmosemsft** [comment]
In practice do our supported systems support a wider range?

**Comment by JeremyKuhne** [comment]
> In practice do our supported systems support a wider range?

Yes, this won't be the only API proposal I'll be opening.

**Comment by jkotas** [comment]
Is there somebody (our own CLI?) eager to use this API for user input validation? Do they plan to issue a warning, or fail with error?

I think that the right policy for cross-platform filenames depends on your tolerance for how often things do not work. I do not think there is one size fits all. For example, it is very common for spaces in filenames or non-ascii characters in filenames to be non-portable either.

**Comment by JeremyKuhne** [comment]
> I think that the right policy for cross-platform filenames depends on your tolerance for how often things do not work.  I do not think there is one size fits all.

True, but this is a standard that "fits almost all". #27049 is the one that hits the normal "I want to create filenames that will work well, even if the location is shared between .NET supported platforms.".  Right now we expose InvalidPathChars which really isn't helpful for cross plat.
